wordpress-mobile/WordPress-iOS
# Site Settings in SwiftUI: General

**HACK week:** update Site Settings to use SwiftUI

Previous PR: #20838

This PR adds all of the fields from the "General" section to the new Site Settings screen.

- Add the remaining fields to the General section: address, privacy, etc
- Introduce `SettingsPicker` to replace `SettingsSelectionViewController`
- Add support for copying site URL (new feature requested in #15927)

<img width="302" alt="Screenshot 2023-06-14 at 12 08 00 PM" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/3f0cf491-750f-4bae-ad30-62576002cc2c">

## To test: 

- Add to AppDelegate:

```swift
DispatchQueue.main.asyncAfter(deadline: .now() + .seconds(1)) {
    let host = NewSiteSettingsViewController(blog: {
        let context = ContextManager.shared.mainContext
        return Blog.lastUsedOrFirst(in: context)!
    }())

    let nav = UINavigationController(rootViewController: host)
    nav.modalPresentationStyle = .fullScreen
    self.window?.rootViewController?.present(nav, animated: true)
}
```

- Change any of the values in the "General" section
- Test that the values are updated (use a different app or the existing settings screen)
- Verify that the language picker works the same way as in the production version (make sure to check search!)

## Regression Notes

1. Potential unintended areas of impact: Site Settings screen
2. What I did to test those areas of impact (or what existing automated tests I relied on): manual tests
3. What automated tests I added (or what prevented me from doing so): n/a

PR submission checklist:


UI Changes testing checklist:
